Src/WitsmlExplorer.Api/HttpHandlers/AuthorizeHandler.cs

```diff
@@ -12,36 +12,38 @@ namespace WitsmlExplorer.Api.HttpHandlers
 {
     public static class AuthorizeHandler
     {
-        public static async Task<IResult> Authorize([FromQuery(Name = "keep")] bool keep, [FromServices] ICredentialsService credentialsService, HttpContext httpContext)
+        public static async Task<IResult> Authorize([FromQuery(Name = "keep")] bool keep, [FromServices] ICredentialsService credentialsService, IConfiguration configuration, HttpContext httpContext)
         {
-            string basicAuth = httpContext?.Request.Headers[EssentialHeaders.WitsmlTargetServer].ToString();
+            bool useOAuth2 = StringHelpers.ToBoolean(configuration[ConfigConstants.OAuth2Enabled]);
+            EssentialHeaders eh = new(httpContext?.Request);
+            string basicAuth = eh.TargetServer;
             ServerCredentials creds = await credentialsService.GetCredentialsFromHeaderValue(basicAuth);
-
+            await credentialsService.VerifyCredentials(creds);
```

Comment:
Should the exception thrown in VerifyCredentials by witsmlClient.TestConnectionAsync() be handled here to give the user a reasonable error message in frontend?

Reply:
That might be a good idea yes to catch and give a 401 response. I will check thanks

Reply:
Seems like it is already being handled by catching an `MessageSecurityException` in `ExceptionMiddleware` giving 401 with an ok message to the end user.

Something similar could perhaps be used for the handlers missing valid creds in cache 🤔

```
                ErrorDetails errorDetails = new()
                {
                    StatusCode = (int)HttpStatusCode.Unauthorized,
                    Message = "Not able to authenticate to WITSML server with given credentials"
                };
                ```

Reply:
Removed the `HasClient`. Added handling in Middleware, thanks